app/tests/algorithms_tests/test_forms.py

```diff
@@ -811,6 +812,50 @@ def test_algorithm_form_gpu_limited_choices():
     assert actual_choices == expected_choices
 
 
+@pytest.mark.django_db
+def test_algorithm_form_gpu_choices_from_phases():
+    user = UserFactory()
+    p1 = PhaseFactory(
+        submission_kind=SubmissionKindChoices.ALGORITHM,
+        algorithm_selectable_gpu_type_choices=[
+            GPUTypeChoices.NO_GPU,
+            GPUTypeChoices.T4,
```

Comment:
I think this and the other tests only test the logic partially. They are missing inputs and outputs.

Reply:
Thanks, you are right. (of course :angel:) 
Updated them

Reply:
Looks good! For a complete test, I think you would also want a phase that has, for example, 2 inputs (`ci_in` and another one) to make sure the `distinct` in the `Count` annotation does its job.

Reply:
Good point. Added it, thanks!